lib/util/pssh.js

```diff
@@ -51,8 +52,14 @@ shaka.util.Pssh = function(psshBox) {
   * */
   this.dataBoundaries = [];
 
-  new shaka.util.Mp4Parser()
+  if (extractMoov) {
```

Comment:
Why make this optional? Why not always look at the moov?

Reply:
this method is originally used with the Ms polyfill that fires encrypted events when it sees a pssh box, however Ms apparently does not extract the pssh completely. The MS PSSH data is however already extracted from the Moov box.

When we manually parse the binary segments the PSSH data is still contained within the Moov box so we need to extract it first, this is why we used the optional argument to still be able to support both use cases. 
 so we don't need to parse it.